examples/08_distributed_manager.py

```diff
@@ -0,0 +1,220 @@
+is a utility that provides a useful set of properties that pertain to a parallel
+environment.
+
+In this example you will learn:
+
+- How to use the distributed manager to access parallel environment properties
+- Parallelize deterministic inference across multiple initial date-times
+- Limitations of parallel inference in Earth2Studio
+- Custom IO for sub-selection of variables
+- Post-processing strategies of parallel job outputs
+"""
+
+# %%
+# Set Up
+# ------
+# Set up the distributed manager by initializing it. Out of the box, the distributed
+# manager supports MPI, SLURM and PyTorch parallel environments which provide information
+# regarding the parallel enviroment but environment variables.
+#
+# For example, this script could be ran using:
+#
+# .. code-block:: bash
+#
+#   mpirun -np 2 python 08_distributed_manager
+#
+# to run inference on two devices.
+
+# %%
+import numpy as np
+import torch
+from dotenv import load_dotenv
+
+load_dotenv()  # TODO: make common example prep function
+from loguru import logger
+from modulus.distributed import DistributedManager
+
+DistributedManager.initialize()  # Only call this once in the entire script!
+dist = DistributedManager()
+assert (  # noqa: S101
+    dist._distributed
+), "Looks like torch distributed isn't set up. Check your env variables!"
+
+logger.info(
+    f"Inference runner {dist.rank} of {dist.world_size} with device {dist.device}"
+)
+# %%
+# Next the needed components get initialized.
+# Rigorous parallel support is not part of Earth2Studio's design goals, there are some
+# spots where potential race conditions can occur. Thus some additional care should be
+# taken to ensure safe parallel inference.
+
+# %%
+from earth2studio.data import ARCO
+from earth2studio.io import ZarrBackend
+from earth2studio.models.px import DLWP
+
+# Load model
+package = DLWP.load_default_package()
+if dist.rank == 0:
+    model = DLWP.load_model(package)
+
+torch.distributed.barrier()
+if dist.rank != 0:
+    model = DLWP.load_model(package)
+
+# %%
+# When loading models that are built into Earth2Studio, the model's checkpoint files
+# will be downloaded into the machines cache. If each inference process has access to
+# the same cache location, then only one should download the checkpoint triggered by
+# :py:func:`load_model`.
+#
+# Here :py:class:`earth2studio.models.px.DLWP` checkpoint files are first downloaded by
+# process 0 and then loaded by other processes.
+
+# %%
+
+# Create the data source
+data = ARCO()
+
+# %%
+# The remote date store will place cached data into separate caches for each process.
+# This makes the download of initial state data safe during parallel inference but also
+# means that multiple jobs will download the same date-time if needed.
+
+# %%
+from typing import Any
+
+from earth2studio.utils.type import CoordSystem
+
+
+# Create the IO handler, just save two variables
+class CustomIO(ZarrBackend):
```

Comment:
Thoughts about this?

I think its an interesting solution but I'm not sure I entirely like it.

Reply:
I do not believe we should support this type of user behavior. For this particular functionality I think it is preferable to ammend `deterministic` with the `ensemble` behavior of allowing passed `output_coords` and using `map_coords` internally.

Reply:
Makes sense to me.

Reply:
Okay added the logic from the ensemble workflow to the others for output coords.
Also added dedicated tests for this functionality for all three.